src/pages/workspaces/workspace/notebooks/NotebookLauncher.js

```diff
@@ -32,32 +74,250 @@ const NotebookLauncher = _.flow(
+  const [locked, setLocked] = useState(false)
+  const [lockedBy, setLockedBy] = useState(null)
+  const [exportingNotebookName, setExportingNotebookName] = useState()
+  const [copyingNotebookName, setCopyingNotebookName] = useState()
+  const clusterStatus = cluster && cluster.status
+  const notebookLink = Nav.getLink('workspace-notebook-launch', { namespace, name, notebookName })
+
+  const checkIfLocked = withErrorReporting('Error checking notebook lock status', async () => {
+    const { metadata: { lastLockedBy, lockExpiresAt } } = await Ajax(signal).Buckets.notebook(namespace, bucketName, notebookName.slice(0, -6)).getObject()
+    const hashedUser = await notebookLockHash(bucketName, email)
+    const lockExpirationDate = new Date(parseInt(lockExpiresAt))
+
+    if (lastLockedBy && (lastLockedBy !== hashedUser) && (lockExpirationDate > Date.now())) {
+      setLocked(true)
+      setLockedBy(lastLockedBy)
+    }
+  })
+
+  Utils.useOnMount(() => { checkIfLocked() })
+
+  const clusterActions = (mode, clusterStatus) => {
+    const choosePlaygroundMode = () => Nav.history.push({
+      search: '?' + qs.stringify({ playground: 'true' })
+    })
+    const chooseEditMode = () => Nav.history.push({
+      search: '?' + qs.stringify({ edit: 'true' })
+    })
+    if (mode === 'playground') {
+      choosePlaygroundMode()
+    } else {
+      chooseEditMode()
+    }
+  }
+
+  return div({ style: { display: 'flex', alignItems: 'center', borderBottom: `2px solid ${colors.dark(0.2)}` } }, [
+    div({ style: { fontSize: 18, fontWeight: 'bold', backgroundColor: colors.dark(0.2), paddingRight: '4rem', paddingLeft: '4rem', height: '100%', display: 'flex', alignItems: 'center' } },
+      ['PREVIEW (READ-ONLY)']),
     Utils.cond(
-      [!canCompute, () => h(ButtonOutline, { onClick: () => setCopying(true) }, ['copy to another workspace to edit'])],
-      () => h(ButtonOutline, {
-        href: Nav.getLink('workspace-notebook-launch', { namespace, name, notebookName })
-      }, ['edit in Jupyter'])
+      [
+        !readOnlyAccess && clusterStatus === 'Creating', () => h(StatusMessage, { showSpinner: true }, [
+          'Creating notebook runtime environment, this will take 5-10 minutes. You can navigate away and return when it’s ready.'
+        ])
+      ],
+      [
+        !readOnlyAccess && clusterStatus === 'Starting', () => h(StatusMessage, { showSpinner: true }, [
+          'Starting notebook runtime environment, this may take up to 2 minutes.'
+        ])
+      ],
+      [
+        !readOnlyAccess && clusterStatus === 'Stopping', () => h(StatusMessage, { showSpinner: true }, [
+          'Notebook runtime environment is stopping. You can restart it after it finishes.'
+        ])
+      ],
+      [!readOnlyAccess && clusterStatus === 'Error', () => h(StatusMessage, ['Notebook runtime error.'])],
+      [
+        !readOnlyAccess && (clusterStatus === 'Running' || clusterStatus === 'Stopped' || clusterStatus === null),
```

Comment:
If your cluster happens to be transitioning when you click into the preview page, you have to wait for it to finish before you are presented with the menu. Since we already provide a status indicator via the `ClusterManager` up top, perhaps we should only show the status messages while in playground or edit mode.